boskos/client/client.go

```diff
@@ -0,0 +1,224 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"k8s.io/test-infra/boskos/common"
+)
+
+var (
+	defaultURL = "http://boskos"
+)
+
+// Public Boskos client object
+type Client struct {
+	url       string
+	resources []string
+	owner     string
+}
+
+// NewClient creates a boskos client, with boskos url and owner of the client.
+func NewClient(url string, owner string) *Client {
+	if url == "" {
```

Comment:
Seems kind of awkward...

I would a) put url after owner in the signature and b) removing url from NewClient entirely